diff --git a/redis-sharding.mlb b/redis-sharding.mlb
index bbe5f63..8f3a415 100644
--- a/redis-sharding.mlb
+++ b/redis-sharding.mlb
@@ -3,7 +3,7 @@ local
 	$(SML_LIB)/basis/basis.mlb
 in
 	ev/hash-array-int.sml
-	ev/ev-kqueue-mlton.sml
+	ev/ev-epoll-mlton.sml
 	sparcl/sparcl-string.sml
 	redis-parser.sml
 	local
diff --git a/redis-sharding.mlp b/redis-sharding.mlp
index 3e7d6f7..6d0b4e3 100644
--- a/redis-sharding.mlp
+++ b/redis-sharding.mlp
@@ -1,5 +1,5 @@
 use "ev/hash-array-int.sml";
-use "ev/ev-kqueue-poly.sml";
+use "ev/ev-epoll-poly.sml";
 use "sparcl/sparcl-string.sml";
 use "redis-parser.sml";
 use "os-constants.sml";
diff --git a/with-poly.sml b/with-poly.sml
index c75be88..68dd453 100644
--- a/with-poly.sml
+++ b/with-poly.sml
@@ -11,7 +11,7 @@ fun main_wrapper f x = (
 
 local 
   open Foreign
-  val libz = loadLibrary "libz.so.6"
+  val libz = loadLibrary "libz.so.1"
   val crc32_zlib_ffi = buildCall3 ((getSymbol libz "crc32"), (cUlong, cString, cUint), cUlong)
 in
   fun crc32_zlib (s:string) : Word32.word = Word32.fromInt (crc32_zlib_ffi (0, s, String.size s))
